java-sdk/radar-schemas-registration/src/main/java/org/radarbase/schema/registration/KafkaTopics.java

```diff
@@ -193,40 +204,42 @@ public boolean createTopics(Stream<String> topicsToCreate, int partitions, short
     public boolean refreshTopics() throws InterruptedException {
         ensureInitialized();
         logger.info("Waiting for topics to become available.");
-        int sleep = 10;
+        int sleep = 2;
+        long timeout = 5;
         int numTries = 10;
 
         topics = null;
-        ListTopicsOptions opts = new ListTopicsOptions().listInternal(true);
+        ListTopicsOptions opts = new ListTopicsOptions()
+                .listInternal(true);
+
         for (int tries = 0; tries < numTries; tries++) {
+            long start = System.currentTimeMillis();
             try {
-                topics = getKafkaClient().listTopics(opts).names().get(sleep, TimeUnit.SECONDS);
-            } catch (ExecutionException e) {
-                logger.error("Failed to list topics from brokers: {}."
-                        + " Trying again after {} seconds.", e, sleep);
-                Thread.sleep(sleep * 1000L);
-                sleep = Math.min(MAX_SLEEP, sleep * 2);
-                continue;
-            } catch (TimeoutException e) {
-                // do nothing
+                topics = getKafkaClient()
+                        .listTopics(opts)
+                        .names()
+                        .get(sleep, TimeUnit.SECONDS);
+            } catch (ExecutionException ex) {
+                logger.error("Failed to list topics from brokers: {}", ex.getCause().toString());
+            } catch (TimeoutException ex) {
+                logger.error("Failed to list topics within {} seconds", timeout);
             }
             if (topics != null && !topics.isEmpty()) {
                 break;
-            }
-            if (tries < numTries - 1) {
-                logger.warn("Topics not listed yet after {} seconds", sleep);
+            } else if (tries < numTries - 1) {
+                logger.warn("Topics not listed yet. Trying again after {} seconds.", sleep);
+                long timeTaken = System.currentTimeMillis() - start;
+                long sleepMillis = Math.max(sleep * 1000L - timeTaken, 0L);
+                if (sleepMillis > 0L) {
+                    Thread.sleep(sleepMillis);
+                }
+                sleep = Math.min(MAX_SLEEP, sleep * 2);
+                timeout = Math.min(5, sleep);
```

Comment:
can we just use sleep or is this min necessary?

Reply:
Indeed, done.